File: crates/smolpc-assistant-types/src/assistant.rs
use crate::mode::AppMode;
use crate::provider::ToolExecutionResultDto;
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct AssistantMessageDto {
    pub role: String,
    pub content: String,
}

#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct AssistantSendRequestDto {
    pub mode: AppMode,
    pub chat_id: Option<String>,
    pub messages: Vec<AssistantMessageDto>,
    pub user_text: String,
}

#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
#[serde(rename_all = "camelCase")]
pub struct AssistantResponseDto {
    pub reply: String,
    pub explain: Option<String>,
    pub undoable: bool,
    pub plan: Option<serde_json::Value>,
    pub tool_results: Vec<ToolExecutionResultDto>,
}

File: app/src/lib/types/assistant.ts
import type { AppMode } from '$lib/types/mode';
import type { ToolExecutionResultDto } from '$lib/types/provider';

export interface AssistantMessageDto {
	role: string;
	content: string;
}

export interface AssistantSendRequestDto {
	mode: AppMode;
	chatId: string | null;
	messages: AssistantMessageDto[];
	userText: string;
}

export interface AssistantResponseDto {
	reply: string;
	explain: string | null;
	undoable: boolean;
	plan: unknown;
	toolResults: ToolExecutionResultDto[];
}

export type AssistantStreamEvent =
	| { kind: 'status'; phase: string; detail: string }
	| { kind: 'tool_call'; name: string; arguments: unknown }
	| { kind: 'tool_result'; name: string; result: ToolExecutionResultDto }
	| { kind: 'token'; token: string }
	| { kind: 'complete'; response: AssistantResponseDto }
	| { kind: 'error'; code: string; message: string };
